spring-boot-dixit/src/test/java/tw/wally/dixit/DixitControllerTest.java

```diff
@@ -0,0 +1,277 @@
+        tellStory(dixit.getNumberOfRounds(), guesser)
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void GivenDixitStartedAndFirstRoundStarted_WhenStorytellerTellStoryAtSecondRound_ThenShouldFail() throws Exception {
+        Dixit dixit = createDixitWithGamersAndGet(6);
+        assertEquals(1, dixit.getNumberOfRounds());
+
+        Player storyteller = dixit.getCurrentStoryteller();
+        tellStory(2, storyteller)
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void GivenStoryTold_WhenTwoGuessersPlayCard_ThenCurrentRoundShouldHaveTwoPlayCards() throws Exception {
+        Dixit dixit = givenStoryToldAndGetDixit();
+
+        int numberOfRounds = dixit.getNumberOfRounds();
+        var guessers = limit(dixit.getCurrentGuessers(), 2);
+        eachGuesserPlayCard(numberOfRounds, guessers);
+
+        assertEachGuesserPlayCard(guessers);
+    }
+
+    @Test
+    public void GivenStoryTold_WhenStorytellerPlayCard_ThenShouldFail() throws Exception {
+        Dixit dixit = givenStoryToldAndGetDixit();
+
+        int numberOfRounds = dixit.getNumberOfRounds();
+        Player storyteller = dixit.getCurrentStoryteller();
+        playCard(numberOfRounds, storyteller)
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void GivenStoryToldAtFirstRound_WhenOneGuesserPlayCardAtSecondRound_ThenShouldFail() throws Exception {
+        Dixit dixit = givenStoryToldAndGetDixit();
+        assertEquals(1, dixit.getNumberOfRounds());
+
+        Player guesser = dixit.getCurrentGuessers().get(0);
+        playCard(2, guesser)
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void GivenEachGuesserPlayedCard_WhenTwoGuessersGuessStory_ThenCurrentRoundShouldHaveTwoGuesses() throws Exception {
+        Dixit dixit = givenEachGuesserPlayedCardAndGetDixit();
+
+        int numberOfRounds = dixit.getNumberOfRounds();
+        var guessers = limit(dixit.getCurrentGuessers(), 2);
+        eachGuesserGuessStory(numberOfRounds, guessers);
+
+        assertEachGuesserGuessStory(guessers);
+    }
+
+    @Test
+    public void GivenEachGuesserPlayedCard_WhenStorytellerGuessStory_ThenShouldFail() throws Exception {
+        Dixit dixit = givenEachGuesserPlayedCardAndGetDixit();
+
+        int numberOfRounds = dixit.getNumberOfRounds();
+        Player storyteller = dixit.getCurrentStoryteller();
+        guessStory(numberOfRounds, storyteller)
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void GivenEachGuesserPlayedCardAtFirstRound_WhenOneGuesserGuessStoryAtSecondRound_ThenShouldFail() throws Exception {
+        Dixit dixit = givenEachGuesserPlayedCardAndGetDixit();
+        assertEquals(1, dixit.getNumberOfRounds());
+
+        var guesser = dixit.getCurrentGuessers().get(0);
+        guessStory(2, guesser)
+                .andExpect(status().isBadRequest());
+    }
```

Comment:
rename to `GivenAllGuessersPlayedCardInFirstRound_WhenOneGuesserGuessStoryInSecondRound_ThenShouldRespondBadRequest`